cmd/template_register.go

```diff
@@ -117,15 +141,15 @@ func templateRegister(registerTemplate egoscale.RegisterCustomTemplate, zone str
 }
 
 func init() {
-	templateRegisterCmd.Flags().StringP("checksum", "", "", "the MD5 checksum value of the template")
-	templateRegisterCmd.Flags().StringP("description", "", "", "the template description")
-	templateRegisterCmd.Flags().StringP("name", "", "", "the name of the template")
+	templateRegisterCmd.Flags().StringP("checksum", "c", "", "the MD5 checksum value of the template")
+	templateRegisterCmd.Flags().StringP("description", "d", "", "the template description")
 	templateRegisterCmd.Flags().StringP("zone", "z", "", "the ID or name of the zone the template is to be hosted on")
-	templateRegisterCmd.Flags().StringP("url", "", "", "the URL of where the template is hosted")
-	templateRegisterCmd.Flags().BoolP("disable-password", "", false, "true if the template does not support password authentication; default is false")
-	templateRegisterCmd.Flags().BoolP("disable-ssh-key", "", false, "true if the template does not support ssh key authentication; default is false")
-	templateRegisterCmd.Flags().StringP("username", "", "", "The default username of the template")
-	templateRegisterCmd.Flags().StringP("boot-mode", "", "legacy", "The template boot mode (legacy/uefi)")
+	templateRegisterCmd.Flags().StringP("username", "u", "", "The default username of the template")
+	templateRegisterCmd.Flags().String("url", "", "the URL of where the template is hosted")
+	templateRegisterCmd.Flags().String("boot-mode", "legacy", "The template boot mode (legacy/uefi)")
+	templateRegisterCmd.Flags().String("from-snapshot", "", "the ID of a snapshot to export it and then reuse the returned snapshot file URL and checksum to register a new custom template")
```

Comment:
No need to explain how the sausage is made 😉 

```suggestion
	templateRegisterCmd.Flags().String("from-snapshot", "", "the ID of a Compute instance snapshot to register as a new template")
```